### DISCUSSION TITLE
Commercial EmguCV 4.9.0.5494 Bitmap extension class is missing

### DISCUSSION THREAD

**Discussion by EkaterinaRi** [emgucv/emgucv#961] [Q&A] [answered]
Hello,

Sorry for a silly question:
I have just started using a commercial version of EmguCV and cannot make any of the Bitmap extension class methods work in my code. 
I am targeting .NET Framework 4.7.2 platform x64. In a free version Emgu.CV.Bitmap is a separate package which is easy to add to references. However, after reading some issues related to a commercial version, I came to a conclusion that .AsBitmap and other Bitmap extension class methods that I need are implemented in the Emgu.CV.UI which should be part of the Emgu.CV.runtime.windows.cuda.dldt that I reference. Nevertheless, I cannot compile my code while I'm using .AsBitmap() and similar, but when I comment it out the project builds without errors.
I have referenced Emgu.CV.runtime.windows.cuda.dldt 4.9.0.5494 (from a commercial nuget repo). The following methods cause errors for me: Image<>.AsBitmap(), Image<>.ToBitmap() or BitmapExtension.ToImage<,>. 
I get the error message: 
CS1061 'Image<Gray, byte>' does not contain a definition for 'ToBitmap' and no accessible extension method 'ToBitmap' accepting a first argument of type 'Image<Gray, byte>' could be found (are you missing a using directive or an assembly reference?).
A "using Emgu.CV.UI;" directive causes the following error message: 
CS0234	The type or namespace name 'UI' does not exist in the namespace 'Emgu.CV' (are you missing an assembly reference?)

What am I doing wrong?

**Comment by emgucv** [comment, marked as answer]
For both commercial and open source runtime, if you want to use the Bitmap extension, you will need to install the Emgu.CV.Bitmap nuget package from nuget.org:
https://www.nuget.org/packages/Emgu.CV.Bitmap
If you look at the package's license terms, it is used by both open source and commercial license. 

It is under Emgu.CV.BitmapExtension class (Not Emgu.CV.UI namespace): 
https://www.emgu.com/wiki/files/4.9.0/document/html/M_Emgu_CV_BitmapExtension_ToBitmap__2_1.htm

Once you added the Emgu.CV.Bitmap nuget package, you can access the method by calling:
```
Image<Gray, byte> img = ...
img.ToBitmap();
```

**Reply by EkaterinaRi** [reply]
Thank you very much!
